src/core/or/connection_edge.c

```diff
@@ -863,18 +863,35 @@ export_hs_client_circuit_id(edge_connection_t *edge_conn,
   if (protocol != HS_CIRCUIT_ID_PROTOCOL_HAPROXY)
     return;
 
+  origin_circuit_t *circ;
```

Comment:
I would recommend using a `const` here and then `CONST_TO_ORIGIN_CIRCUIT()` for extra safety.

Reply:
Fixed in `b0e2baf`.